grml2usb

```diff
@@ -64,181 +63,185 @@ RE_P_PARTITION = re.compile(r"(.*?\d+)p(\d+)$")
-parser.add_option(
+parser.add_argument(
     "--quiet",
-    dest="quiet",
     action="store_true",
     help="do not output anything but just errors on console",
 )
-parser.add_option(
+parser.add_argument(
     "--remove-bootoption",
     dest="removeoption",
     action="append",
     help="regex for removing existing bootoptions",
 )
-parser.add_option(
+parser.add_argument(
     "--rw-blockdev",
     dest="rwblockdev",
     action="store_true",
     help="enforce read-write mode on involved block devices",
 )
-parser.add_option(
+parser.add_argument(
     "--skip-addons",
     dest="skipaddons",
     action="store_true",
     help="do not install /boot/addons/ files",
 )
-parser.add_option(
+parser.add_argument(
     "--skip-bootflag",
     dest="skipbootflag",
     action="store_true",
     help="do not try to check whether the destination has the boot flag set",
 )
-parser.add_option(
+parser.add_argument(
     "--skip-grub-config",
     dest="skipgrubconfig",
     action="store_true",
     help="skip generation of grub configuration files",
 )
-parser.add_option(
+parser.add_argument(
     "--skip-mbr",
     dest="skipmbr",
     action="store_true",
     help="do not install a master boot record (MBR) on the device",
 )
-parser.add_option(
+parser.add_argument(
     "--skip-syslinux-config",
     dest="skipsyslinuxconfig",
     action="store_true",
     help="skip generation of syslinux configuration files",
 )
-parser.add_option(
+parser.add_argument(
     "--skip-usb-check",
     dest="skipusbcheck",
     action="store_true",
     help="skip check to verify whether given device is removable",
 )
-parser.add_option(
-    "--syslinux",
-    dest="syslinux",
-    action="callback",
-    default=True,
-    callback=syslinux_warning,
-    help="install syslinux bootloader (deprecated as it's the default)",
-)
-parser.add_option(
+parser.add_argument(
     "--syslinux-mbr",
     dest="syslinuxmbr",
     action="store_true",
     help="install syslinux master boot record (MBR) instead of default",
 )
-parser.add_option(
+parser.add_argument(
     "--syslinux-libs",
     dest="syslinuxlibs",
     action="append",
     default=[],
     help="syslinux modules directory path",
 )
-parser.add_option(
+parser.add_argument(
     "--tmpdir",
-    dest="tmpdir",
     default="/tmp",
     help="directory to be used for temporary files",
 )
-parser.add_option(
-    "--verbose", dest="verbose", action="store_true", help="enable verbose mode"
+parser.add_argument(
+    "--verbose",
+    action="store_true",
+    help="enable verbose mode",
 )
-parser.add_option(
-    "-v",
+parser.add_argument(
     "--version",
-    dest="version",
-    action="store_true",
+    "-v",
+    action="version",
+    version=f"%(prog)s {PROG_VERSION}",
     help="display version and exit",
 )
-(options, args) = parser.parse_args()
+
+bootloader_group.add_argument(
+    "--grub",
+    action="store_true",
+    help="install grub bootloader instead of (default) syslinux",
+)
```

Comment:
your idea here is to still support the `--grub` and `--syslinux` options but no longer announce them via `--help`?

Reply:
They are still in `--help`, but as a mutually exclusive group. With this `argparse` takes care of preventing the user to set both.

Reply:
Ah I see, thanks!